src/otx/core/data/pre_filtering.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+"""Pre filtering data for OTX."""
+
+from __future__ import annotations
+
+import warnings
+from typing import TYPE_CHECKING
+
+from datumaro.components.annotation import Annotation, Bbox, Polygon
+from datumaro.components.dataset import Dataset as DmDataset
+
+if TYPE_CHECKING:
+    from datumaro.components.dataset_base import DatasetItem
+
+
+def pre_filtering(dataset: DmDataset, data_format: str) -> DmDataset:
+    """Filtering invalid data in datumaro dataset."""
+    dataset = DmDataset.filter(dataset, is_non_empty_item)
+    dataset = DmDataset.filter(dataset, is_valid_annot, filter_annotations=True)
+
+    return remove_unused_labels(dataset, data_format)
+
+
+def is_non_empty_item(item: DatasetItem) -> bool:
+    """Return whether DatasetItem's annotation is non-empty."""
+    if not (item.subset == "train" and len(item.annotations) == 0):
+        return True
+    msg = "There are empty annotation items in train set, they will be filtered out before training."
+    warnings.warn(msg, stacklevel=2)
+    return False
+
+
+def is_valid_annot(item: DatasetItem, annotation: Annotation) -> bool:  # noqa: ARG001
+    """Return whether DatasetItem's annotation is valid."""
+    if isinstance(annotation, Bbox):
+        x1, y1, x2, y2 = annotation.points
+        if x1 < x2 and y1 < y2:
+            return True
+        msg = "There are bounding box which is not `x1 < x2 and y1 < y2`, they will be filtered out before training."
+        warnings.warn(msg, stacklevel=2)
+        return False
+    if isinstance(annotation, Polygon):
+        # TODO(JaegukHyun): This process is computationally intensive.  # noqa: TD003
+        # We should make pre-filtering user-configurable.
+        x_points = [annotation.points[i] for i in range(0, len(annotation.points), 2)]
+        y_points = [annotation.points[i + 1] for i in range(0, len(annotation.points), 2)]
+        if min(x_points) < max(x_points) and min(y_points) < max(y_points) and annotation.get_area() > 0:
+            return True
+        msg = "There are invalid polygon, they will be filtered out before training."
+        return False
+    return True
+
+
+def remove_unused_labels(dataset: DmDataset, data_format: str) -> DmDataset:
+    """Remove unused labels in Datumaro dataset."""
+    original_categories: list[str] = dataset.get_label_cat_names()
+    used_labels: list[int] = list({ann.label for item in dataset for ann in item.annotations})
+    if data_format == "ava":
```

Comment:
Not want to apply to this PR, however, I just hope to make the explicit `enum` or `class` to represent the data format. 
```python3
class OTXDataFormat(str, Enum):
 AVA_DATASET = "ava"
...
```

```

Reply:
Sounds great, Thanks for the recommendation